[gazebo_plugins] Build fail: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’

With today's updates on `indigo-devel`, I get the following build error:
(I'm using regular deb for gazebo2 `2.2.3-1~trusty`)

```
Scanning dependencies of target gazebo_ros_skid_steer_drive
[  9%] Building CXX object gazebo_ros_pkgs/gazebo_plugins/CMakeFiles/gazebo_ros_skid_steer_drive.dir/src/gazebo_ros_skid_steer_drive.cpp.o
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp: In member function ‘virtual void gazebo::GazeboRosSkidSteerDrive::Load(gazebo::physics::ModelPtr, sdf::ElementPtr)’:
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:259:25: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
     joints[LEFT_FRONT]->SetParam("fmax", 0, torque);
                         ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:260:26: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
     joints[RIGHT_FRONT]->SetParam("fmax", 0, torque);
                          ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:261:24: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
     joints[LEFT_REAR]->SetParam("fmax", 0, torque);
                        ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:262:25: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
     joints[RIGHT_REAR]->SetParam("fmax", 0, torque);
                         ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp: In member function ‘virtual void gazebo::GazeboRosSkidSteerDrive::UpdateChild()’:
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:311:27: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
       joints[LEFT_FRONT]->SetParam("vel", 0, wheel_speed_[LEFT_FRONT] / (wheel_diameter_ / 2.0));
                           ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:312:28: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
       joints[RIGHT_FRONT]->SetParam("vel", 0, wheel_speed_[RIGHT_FRONT] / (wheel_diameter_ / 2.0));
                            ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:313:26: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
       joints[LEFT_REAR]->SetParam("vel", 0, wheel_speed_[LEFT_REAR] / (wheel_diameter_ / 2.0));
                          ^
/home/fxm/projects/indigo/control/src/gazebo_ros_pkgs/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp:314:27: error: ‘class gazebo::physics::Joint’ has no member named ‘SetParam’
       joints[RIGHT_REAR]->SetParam("vel", 0, wheel_speed_[RIGHT_REAR] / (wheel_diameter_ / 2.0));
                           ^
make[2]: *** [gazebo_ros_pkgs/gazebo_plugins/CMakeFiles/gazebo_ros_skid_steer_drive.dir/src/gazebo_ros_skid_steer_drive.cpp.o] Error 1
make[1]: *** [gazebo_ros_pkgs/gazebo_plugins/CMakeFiles/gazebo_ros_skid_steer_drive.dir/all] Error 2
make: *** [all] Error 2
make: INTERNAL: Exiting with 5 jobserver tokens available; should be 4!
Invoking "make -j4 -l4" failed
```

--- a/gazebo_plugins/src/gazebo_ros_diff_drive.cpp
+++ b/gazebo_plugins/src/gazebo_ros_diff_drive.cpp
@@ -100,8 +100,13 @@ void GazeboRosDiffDrive::Load ( physics::ModelPtr _parent, sdf::ElementPtr _sdf 
     joints_.resize ( 2 );
     joints_[LEFT] = gazebo_ros_->getJoint ( parent, "leftJoint", "left_joint" );
     joints_[RIGHT] = gazebo_ros_->getJoint ( parent, "rightJoint", "right_joint" );
+#if GAZEBO_MAJOR_VERSION > 2
     joints_[LEFT]->SetParam ( "fmax", 0, wheel_torque );
     joints_[RIGHT]->SetParam ( "fmax", 0, wheel_torque );
+#else
+    joints_[LEFT]->SetMaxForce ( 0, wheel_torque );
+    joints_[RIGHT]->SetMaxForce ( 0, wheel_torque );
+#endif
 
 
 
@@ -170,8 +175,13 @@ void GazeboRosDiffDrive::Reset()
   pose_encoder_.theta = 0;
   x_ = 0;
   rot_ = 0;
+#if GAZEBO_MAJOR_VERSION > 2
   joints_[LEFT]->SetParam ( "fmax", 0, wheel_torque );
   joints_[RIGHT]->SetParam ( "fmax", 0, wheel_torque );
+#else
+  joints_[LEFT]->SetMaxForce ( 0, wheel_torque );
+  joints_[RIGHT]->SetMaxForce ( 0, wheel_torque );
+#endif
 }
 
 void GazeboRosDiffDrive::publishWheelJointState()
@@ -221,8 +231,13 @@ void GazeboRosDiffDrive::UpdateChild()
        (this seems to be solved in https://bitbucket.org/osrf/gazebo/commits/ec8801d8683160eccae22c74bf865d59fac81f1e)
     */
     for ( int i = 0; i < 2; i++ ) {
+#if GAZEBO_MAJOR_VERSION > 2
       if ( fabs(wheel_torque -joints_[i]->GetParam ( "fmax", 0 )) > 1e-6 ) {
         joints_[i]->SetParam ( "fmax", 0, wheel_torque );
+#else
+      if ( fabs(wheel_torque -joints_[i]->GetMaxForce ( 0 )) > 1e-6 ) {
+        joints_[i]->SetMaxForce ( 0, wheel_torque );
+#endif
       }
     }
 
@@ -248,8 +263,13 @@ void GazeboRosDiffDrive::UpdateChild()
                 ( fabs ( wheel_speed_[LEFT] - current_speed[LEFT] ) < 0.01 ) ||
                 ( fabs ( wheel_speed_[RIGHT] - current_speed[RIGHT] ) < 0.01 ) ) {
             //if max_accel == 0, or target speed is reached
+#if GAZEBO_MAJOR_VERSION > 2
             joints_[LEFT]->SetParam ( "vel", 0, wheel_speed_[LEFT]/ ( wheel_diameter_ / 2.0 ) );
             joints_[RIGHT]->SetParam ( "vel", 0, wheel_speed_[RIGHT]/ ( wheel_diameter_ / 2.0 ) );
+#else
+            joints_[LEFT]->SetVelocity ( 0, wheel_speed_[LEFT]/ ( wheel_diameter_ / 2.0 ) );
+            joints_[RIGHT]->SetVelocity ( 0, wheel_speed_[RIGHT]/ ( wheel_diameter_ / 2.0 ) );
+#endif
         } else {
             if ( wheel_speed_[LEFT]>=current_speed[LEFT] )
                 wheel_speed_instr_[LEFT]+=fmin ( wheel_speed_[LEFT]-current_speed[LEFT],  wheel_accel * seconds_since_last_update );
@@ -264,8 +284,13 @@ void GazeboRosDiffDrive::UpdateChild()
             // ROS_INFO("actual wheel speed = %lf, issued wheel speed= %lf", current_speed[LEFT], wheel_speed_[LEFT]);
             // ROS_INFO("actual wheel speed = %lf, issued wheel speed= %lf", current_speed[RIGHT],wheel_speed_[RIGHT]);
 
+#if GAZEBO_MAJOR_VERSION > 2
             joints_[LEFT]->SetParam ( "vel", 0, wheel_speed_instr_[LEFT] / ( wheel_diameter_ / 2.0 ) );
             joints_[RIGHT]->SetParam ( "vel", 0, wheel_speed_instr_[RIGHT] / ( wheel_diameter_ / 2.0 ) );
+#else
+            joints_[LEFT]->SetVelocity ( 0,wheel_speed_instr_[LEFT] / ( wheel_diameter_ / 2.0 ) );
+            joints_[RIGHT]->SetVelocity ( 0,wheel_speed_instr_[RIGHT] / ( wheel_diameter_ / 2.0 ) );
+#endif
         }
         last_update_time_+= common::Time ( update_period_ );
     }

--- a/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp
+++ b/gazebo_plugins/src/gazebo_ros_skid_steer_drive.cpp
@@ -256,10 +256,17 @@ namespace gazebo {
 	 gzthrow(error);
    }
 
+#if GAZEBO_MAJOR_VERSION > 2
     joints[LEFT_FRONT]->SetParam("fmax", 0, torque);
     joints[RIGHT_FRONT]->SetParam("fmax", 0, torque);
     joints[LEFT_REAR]->SetParam("fmax", 0, torque);
     joints[RIGHT_REAR]->SetParam("fmax", 0, torque);
+#else
+    joints[LEFT_FRONT]->SetMaxForce(0, torque);
+    joints[RIGHT_FRONT]->SetMaxForce(0, torque);
+    joints[LEFT_REAR]->SetMaxForce(0, torque);
+    joints[RIGHT_REAR]->SetMaxForce(0, torque);
+#endif
 
     // Make sure the ROS node for Gazebo has already been initialized
     if (!ros::isInitialized())
@@ -308,10 +315,17 @@ namespace gazebo {
 
       // Update robot in case new velocities have been requested
       getWheelVelocities();
+#if GAZEBO_MAJOR_VERSION > 2
       joints[LEFT_FRONT]->SetParam("vel", 0, wheel_speed_[LEFT_FRONT] / (wheel_diameter_ / 2.0));
       joints[RIGHT_FRONT]->SetParam("vel", 0, wheel_speed_[RIGHT_FRONT] / (wheel_diameter_ / 2.0));
       joints[LEFT_REAR]->SetParam("vel", 0, wheel_speed_[LEFT_REAR] / (wheel_diameter_ / 2.0));
       joints[RIGHT_REAR]->SetParam("vel", 0, wheel_speed_[RIGHT_REAR] / (wheel_diameter_ / 2.0));
+#else
+      joints[LEFT_FRONT]->SetVelocity(0, wheel_speed_[LEFT_FRONT] / (wheel_diameter_ / 2.0));
+      joints[RIGHT_FRONT]->SetVelocity(0, wheel_speed_[RIGHT_FRONT] / (wheel_diameter_ / 2.0));
+      joints[LEFT_REAR]->SetVelocity(0, wheel_speed_[LEFT_REAR] / (wheel_diameter_ / 2.0));
+      joints[RIGHT_REAR]->SetVelocity(0, wheel_speed_[RIGHT_REAR] / (wheel_diameter_ / 2.0));
+#endif
 
       last_update_time_+= common::Time(update_period_);
 

--- a/gazebo_plugins/src/gazebo_ros_tricycle_drive.cpp
+++ b/gazebo_plugins/src/gazebo_ros_tricycle_drive.cpp
@@ -108,8 +108,13 @@ void GazeboRosTricycleDrive::Load ( physics::ModelPtr _parent, sdf::ElementPtr _
     odomOptions["world"] = WORLD;
     gazebo_ros_->getParameter<OdomSource> ( odom_source_, "odometrySource", odomOptions, WORLD );
 
+#if GAZEBO_MAJOR_VERSION > 2
     joint_wheel_actuated_->SetParam ( "fmax", 0, wheel_torque_ );
     joint_steering_->SetParam ( "fmax", 0, wheel_torque_ );
+#else
+    joint_wheel_actuated_->SetMaxForce ( 0, wheel_torque_ );
+    joint_steering_->SetMaxForce ( 0, wheel_torque_ );
+#endif
 
 
     // Initialize update rate stuff
@@ -235,7 +240,11 @@ void GazeboRosTricycleDrive::motorController ( double target_speed, double targe
             applied_speed = current_speed - wheel_deceleration_ * dt;
         }
     }
+#if GAZEBO_MAJOR_VERSION > 2
     joint_wheel_actuated_->SetParam ( "vel", 0, applied_speed );
+#else
+    joint_wheel_actuated_->SetVelocity ( 0, applied_speed );
+#endif
     
     double current_angle = joint_steering_->GetAngle ( 0 ).Radian();
     if(target_angle > +M_PI / 2.0) target_angle =  +M_PI / 2.0;
@@ -249,7 +258,11 @@ void GazeboRosTricycleDrive::motorController ( double target_speed, double targe
         } else {
             applied_steering_speed = -steering_speed_;
         }
+#if GAZEBO_MAJOR_VERSION > 2
       joint_steering_->SetParam ( "vel", 0, applied_steering_speed );
+#else
+      joint_steering_->SetVelocity ( 0, applied_steering_speed );
+#endif
     }else {
 #if GAZEBO_MAJOR_VERSION >= 4
       joint_steering_->SetPosition ( 0, applied_angle );

--- a/gazebo_ros_control/src/default_robot_hw_sim.cpp
+++ b/gazebo_ros_control/src/default_robot_hw_sim.cpp
@@ -219,7 +219,11 @@ bool DefaultRobotHWSim::initSim(
         // joint->SetParam("fmax") must be called if joint->SetAngle() or joint->SetParam("vel") are
         // going to be called. joint->SetParam("fmax") must *not* be called if joint->SetForce() is
         // going to be called.
+#if GAZEBO_MAJOR_VERSION > 2
         joint->SetParam("fmax", 0, joint_effort_limits_[j]);
+#else
+        joint->SetMaxForce(0, joint_effort_limits_[j]);
+#endif
       }
     }
   }
@@ -327,7 +331,11 @@ void DefaultRobotHWSim::writeSim(ros::Time time, ros::Duration period)
         break;
 
       case VELOCITY:
+#if GAZEBO_MAJOR_VERSION > 2
         sim_joints_[j]->SetParam("vel", 0, e_stop_active_ ? 0 : joint_velocity_command_[j]);
+#else
+        sim_joints_[j]->SetVelocity(0, e_stop_active_ ? 0 : joint_velocity_command_[j]);
+#endif
         break;
 
       case VELOCITY_PID:

